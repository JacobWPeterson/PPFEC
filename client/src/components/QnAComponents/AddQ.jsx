import React from 'react';
import styled from 'styled-components';
import Modal from './Modal.jsx';
import FormQ from './FormQuestion.jsx';

const AddAQ = styled.button`
  background-color: #FFFFFF;
  border: 1px solid #424242;
  height: 6vh;
  width: 20vw;
  font-size: 15px;
  color: #424242;
  &:hover {
    cursor: pointer;
    color: #80CCC4;
    border: 1px solid #80CCC4;
    box-shadow: 0 3px 6px #a0a0a0, 0 3px 6px #a0a0a0;
  };
  &:focus {
    outline: none;
    box-shadow: none;
  };
`;

const AQC = styled.div`
  padding: 1vh 2vw 1vh 2vw;
`;

class AddQ extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      modal: false,
    };
    this.handleClick = this.handleClick.bind(this);
    this.exitModal = this.exitModal.bind(this);
  }

  handleClick() {
    this.setState({ modal: true });
  }

  exitModal(data) {
    this.setState({ modal: false }, () => {
      if (data) {
        this.props.func(data);
      }
    });
  }

  render() {
    return (
      <AQC>
        <AddAQ type="submit" onClick={this.handleClick}>ADD A QUESTION +</AddAQ>
        {this.state.modal ? (
          <Modal content={(
            <FormQ
              func={this.exitModal}
            />
          )}
          />
        ) : null}
      </AQC>
    );
  }
}

export default AddQ;
