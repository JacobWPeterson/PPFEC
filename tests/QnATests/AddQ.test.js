import React from 'react';
import { mount } from 'enzyme';
import AddQ from '../../client/src/components/QnAComponents/AddQ.jsx';

describe('This is for the Add Questions', () => {
  const AddQComp = mount(<AddQ />);

  it('Should render Add Question Properly', () => {
    expect(AddQComp).toMatchSnapshot();
  });
});
