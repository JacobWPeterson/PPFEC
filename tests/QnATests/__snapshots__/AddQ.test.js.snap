// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`This is for the Add Questions Should render Add Question Properly 1`] = `
<AddQ>
  <AddQ__AQC>
    <div
      className="AddQ__AQC-fu61ov-1 bKyMVa"
    >
      <AddQ__AddAQ
        onClick={[Function]}
        type="submit"
      >
        <button
<<<<<<< HEAD
          className="AddQ__AddAQ-fu61ov-0 dzUWvS"
=======
          className="AddQ__AddAQ-fu61ov-0 fsXLKI"
>>>>>>> b98853f8bfe7cb666fe12c3fd4e9ca4bd19592b9
          onClick={[Function]}
          type="submit"
        >
          ADD A QUESTION +
        </button>
      </AddQ__AddAQ>
    </div>
  </AddQ__AQC>
</AddQ>
`;
